cli/tools/task.rs

```diff
@@ -222,79 +250,88 @@ fn print_available_tasks(
       "  {}",
       colors::red("No tasks found in configuration file")
     )?;
-  } else {
-    let mut seen_task_names =
-      HashSet::with_capacity(tasks_config.tasks_count());
-    for maybe_config in [&tasks_config.member, &tasks_config.root] {
-      let Some(config) = maybe_config else {
-        continue;
-      };
-      for (is_root, is_deno, (key, task)) in config
-        .deno_json
-        .as_ref()
-        .map(|config| {
-          let is_root = !is_cwd_root_dir
-            && config.folder_url
-              == *workspace_dir.workspace.root_dir().as_ref();
-          config
-            .tasks
-            .iter()
-            .map(move |(k, t)| (is_root, true, (k, Cow::Borrowed(t))))
-        })
-        .into_iter()
-        .flatten()
-        .chain(
-          config
-            .package_json
-            .as_ref()
-            .map(|config| {
-              let is_root = !is_cwd_root_dir
-                && config.folder_url
-                  == *workspace_dir.workspace.root_dir().as_ref();
-              config.tasks.iter().map(move |(k, v)| {
-                (is_root, false, (k, Cow::Owned(Task::Definition(v.clone()))))
-              })
-            })
-            .into_iter()
-            .flatten(),
-        )
-      {
-        if !seen_task_names.insert(key) {
+    return Ok(());
+  }
+
+  struct AvailableTaskDescription {
+    is_root: bool,
+    is_deno: bool,
+    name: String,
+    task: TaskDefinition,
+  }
+  let mut seen_task_names = HashSet::with_capacity(tasks_config.tasks_count());
+  let mut task_descriptions = Vec::with_capacity(tasks_config.tasks_count());
+
+  for maybe_config in [&tasks_config.member, &tasks_config.root] {
+    let Some(config) = maybe_config else {
+      continue;
+    };
+
```

Comment:
I rewrote the bulk of this function. It much easier to read and grep for a tiny bit of duplication. It's not a hot path so using complicated iterator ops here is not worth it.